### PR TITLE
Add --ignore option to deprecation and unused commands

### PR DESCRIPTION
**Pull request by uesteibar**
Fixes https://github.com/sketch-hq/Cloud/issues/16076

Usage:

```
$ gql-lint unused --output json --schema path/to/schema.gql --ignore some/pattern_*.gql path/to/**/*.gql
```

### CODE REVIEW

File: cmd/ops/deprecation.go [hunk 1]
```diff
@@ -18,14 +18,16 @@ var deprecationsCmd = &cobra.Command{
...
+	Args: MinimumNArgs(1, "you must specify at least one file with queries or mutations"),
 	RunE: deprecationsCmdRun,
 }
 
 func init() {
 	Program.AddCommand(deprecationsCmd)
 	deprecationsCmd.Flags().StringVar(&flags.schemaFile, schemaFileFlagName, "", "Server's schema as file or url (required)")
 	deprecationsCmd.MarkFlagRequired(schemaFileFlagName) //nolint:errcheck // will err if flag doesn't exist
+
+	deprecationsCmd.Flags().StringArrayVar(&flags.ignore, ignoreFlagName, []string{}, "Files to ignore (as file blob)")
```
Comment (omh) [inline cmd/ops/deprecation.go:30]: Not quite sure what "as a blob" means here...? 🤔 I think maybe "Files to ignore" is enough...

Comment (uesteibar) [reply cmd/ops/deprecation.go:30]: ups, sorry, I meant "glob" 😄

File: input/input.go [hunk 2]
```diff
@@ -72,7 +103,7 @@ func expand(pattern string) ([]string, error) {
 						return err
 					}
 
-					hits = append(hits, path)
+					hits = append(hits, strings.Replace(path, `//`, `/`, -1))
```
Comment (omh) [inline input/input.go:106]: I think you can use `filepath.Clean()` here...

File: input/input_test.go [hunk 3]
```diff
@@ -34,10 +35,17 @@ func TestQueryFiles(t *testing.T) {
...
+		{
+			name:   "expands nested directories",
+			args:   []string{"testdata/**/*.gql"},
+			ignore: []string{"testdata/**/one.gql"},
+			want:   []string{"testdata/two.gql", "testdata/nested/two.gql"},
+		},
 	}
+
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
```
Comment (omh) [inline input/input_test.go:47]: Nice litte table test you got here :D

**Review by omh — APPROVED** [review]
Nice work - just a question and minor nitpick!

File: input/input.go (hunk 2 above)
Comment (uesteibar) [reply input/input.go:106]: oh noice!

**Review by nestor-sk — APPROVED** [review]
🐑 🌈 🦈

File: input/input.go [hunk 4]
```diff
@@ -11,24 +11,55 @@ type command interface {
 	InOrStdin() io.Reader
 }
 
-func QueryFiles(args []string) ([]string, error) {
+func ExpandGlobs(args []string, ignore []string) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (uesteibar) [reply input/input.go:14]: I always struggle to read that format 😬

File: input/input.go [hunk 5]
```diff
@@ -11,24 +11,55 @@ type command interface {
...
+			files = append(files, filter(matches, ignore)...)
+		} else if !contains(ignore, source) {
 			files = append(files, source)
 		}
 	}
 
 	return unique(files), nil
 }
 
+func filter(in []string, ignore []string) []string {
```
[the remark this answers is not in the document]
Comment (uesteibar) [reply input/input.go:41]: yeah that'd be a nice cleanup, absolutely 👍 I agree we should do it separately

[the remark this answers is not in the document]
Comment (omh) [reply input/input.go:41]: My take is why bring in another dependency if this is our only need for sets/unique slices? Yes, a set would be a bit cleaner, but it's another dep with a new api etc that someone touching this code in the future would need to understand or read up on.

**Pull request merged**